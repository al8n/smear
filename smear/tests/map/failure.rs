
#[smear::hashmap(
  key(String),
  value(String)
)]
enum Foo {

}


#[smear::btreemap(
  key(String),
  value(String)
)]
struct Bar(f64, u32);


#[smear::hashmap(
  key = String,
  value("String"),
  attributes(derive(Clone))
)]
struct Baz {
  f1: u64,
  f2: u32,
}

#[smear::btreemap(
  key = "String",
  value(String)
)]
struct Bazz(u64);

fn main() {}