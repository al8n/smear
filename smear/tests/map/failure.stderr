error: Unsupported shape `enum`. Expected struct with no fields.
 --> tests/map/failure.rs:2:1
  |
2 | / #[smear::hashmap(
3 | |   key(String),
4 | |   value(String)
5 | | )]
  | |__^
  |
  = note: this error originates in the attribute macro `smear::hashmap` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Unsupported shape `unnamed fields`. Expected no fields.
  --> tests/map/failure.rs:11:1
   |
11 | / #[smear::btreemap(
12 | |   key(String),
13 | |   value(String)
14 | | )]
   | |__^
   |
   = note: this error originates in the attribute macro `smear::btreemap` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Unsupported shape `named fields`. Expected no fields.
  --> tests/map/failure.rs:18:1
   |
18 | / #[smear::hashmap(
19 | |   key = String,
20 | |   value("String"),
21 | |   attributes(derive(Clone))
22 | | )]
   | |__^
   |
   = note: this error originates in the attribute macro `smear::hashmap` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Unsupported shape `one unnamed field`. Expected no fields.
  --> tests/map/failure.rs:28:1
   |
28 | / #[smear::btreemap(
29 | |   key = "String",
30 | |   value(String)
31 | | )]
   | |__^
   |
   = note: this error originates in the attribute macro `smear::btreemap` (in Nightly builds, run with -Z macro-backtrace for more info)
